### PR TITLE
Add allowed_mentions to interaction responses

### PR DESCRIPTION
**Pull request by 14ROVI**
## Summary
Adds functionality to use allowed mentions in interaction responses (inital and followup responses).

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by TAG-Epic — APPROVED** [review]
Not tested

**Review by DenverCoder1 — COMMENTED** [review]
Tested and working for users

```py
@client.slash_command(guild_ids=[TESTING_GUILD_ID])
async def test_allowed_mentions_true(interaction: Interaction):
    await interaction.send(f"This is a test {interaction.user.mention}", allowed_mentions=AllowedMentions(users=[interaction.user]))

@client.slash_command(guild_ids=[TESTING_GUILD_ID])
async def test_allowed_mentions_false(interaction: Interaction):
    await interaction.send(f"This is a test {interaction.user.mention}", allowed_mentions=AllowedMentions(users=[]))
```
image

I tried testing it but for some reason I can't get mentions to show up when using @ everyone

```py
@client.slash_command(guild_ids=[TESTING_GUILD_ID])
async def test_allowed_mentions_true(interaction: Interaction):
    await interaction.send("This is a test @everyone", allowed_mentions=AllowedMentions(everyone=True))

@client.slash_command(guild_ids=[TESTING_GUILD_ID])
async def test_allowed_mentions_false(interaction: Interaction):
    await interaction.send("This is a test @everyone", allowed_mentions=AllowedMentions(everyone=False))
```

image

The same thing with ext.commands works though

image

Maybe I'm doing something wrong?

**Review by DenverCoder1 — APPROVED** [review]

File: nextcord/interactions.py [hunk 1]
```diff
@@ -407,6 +407,7 @@ async def send(
         tts: bool = False,
         ephemeral: bool = False,
         delete_after: Optional[float] = None,
+        allowed_mentions: Optional[AllowedMentions] = MISSING,
```
Comment (TAG-Epic) [inline nextcord/interactions.py:410]: This & other instances of this should default to None? None does not have a special case here (like resetting nicks) so None should be used?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: nextcord/interactions.py (hunk 1 above)
Comment (14ROVI) [reply nextcord/interactions.py:410]: In the send functions for sync.py and async.py its `allowed_mentions: AllowedMentions = MISSING`. Should I change this to `allowed_mentions: Optional[AllowedMentions] = None` aswell? or just instances where AllowedMentions is already in an optional type.

Comment (TAG-Epic) [reply nextcord/interactions.py:410]: Weirddddd..... lets just leave it as is for now and solve it in v3

**Review by ooliver1 — APPROVED** [review]

**Comment by ooliver1** [comment]
tested

**Pull request merged**